modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/user/mgt/ReadOnlyLDAPUserStoreManagerTestCase.java

```diff
@@ -174,8 +175,9 @@ public void testAddRemoveUsersOfRole() {
         try {
             userMgtClient.addRemoveUsersOfRole(newUserRole, newUsers, deletedUsers);
         } catch (Exception e) {
-            Assert.assertEquals(e.getMessage(), "Read-only user store.Roles cannot be added or modfified"
-                    ,"Error message mismatched");
+            Assert.assertTrue(e.getMessage().contains("Read-only user store.Roles cannot be added or modfified"),
```

Comment:
Addition to this you can use @Test(expectedExceptions = { IOException.class }) annotation in TestNg. And try to avoid catch root exception in test class.